### DISCUSSION TITLE
Incoming relay host and other questions about relays

### DISCUSSION THREAD

**Discussion by d-513** [docker-mailserver/docker-mailserver#2812] [Q&A]
I know docker-mailserver supports relay for outgoing mail. My ISP blocks port 25 so I want to use host a relay using postfix on a VPS.

I have a few questions:
- Does anyone have some example configs for a relay service? Or software I can use to spin that up easier than manually setting up postfix?
- I also would like to setup an incoming mail relay for stability, so if my home is down it will retry sending. I couldn't find much info about this. I just need it to receive SMTP from the VPS and forward to docker-mailserver, right? Is there any special config needed on the docker-mailserver side? Won't it be marked as spam?
- What's the deal with DKIM when using a relay?

**Comment by polarathene** [comment]
You may find [this a helpful reference](https://github.com/docker-mailserver/docker-mailserver/issues/2601#issuecomment-1135520225). I shared some config for testing a local offline setup between several `docker-mailserver` instances for testing relaying.

If you want to relay inbound and outbound, you might as well just have the VPS handle both, rather than relaying back-and-forth to your local system? You'd just use a mail client like Thunderbird to retrieve your mail from the VPS over IMAP, and then submit mail to send from the VPS by Thunderbird connecting over port 465 or 587 (requires login account + password).

Note that many VPS also block port 25 for outbound/sending mail. If that would still be an issue, instead of a VPS, you can use a relay service like SendGrid. Or you can pay like $15 / year (cheaper than a VPS) to use a service like MXRoute which will provide a mail server and webmail service if that suits you. It could also receive the mail, handle spam, and relay to your own mail server if you'd still want that I think.

I don't know enough about DKIM or relays to advise on that.

---

> I also would like to setup an incoming mail relay for stability, so if my home is down it will retry sending. I couldn't find much info about this. I just need it to receive SMTP from the VPS and forward to docker-mailserver, right? Is there any special config needed on the docker-mailserver side? Won't it be marked as spam?

The link I provided will demonstrate how to set that up. There is three DMS instances, the 2nd / middle one demonstrates it relaying from one mail server (sending) to another (receiving), where it's been authorized to send/relay mail on behalf of the sending server.

Do note that you'd need a public internet domain + DNS setup for the mail to be sent to the VPS instance, before it relays to your local presumably private server. If you get the linked reference setup working, you can take down a container that would represent your local mail server, and see how the remote (VPS) one would keep retrying to send IIRC. Then you can adjust the setup to run on different servers once you're happy with the offline setup working the way you want :)

**Reply by d-513** [reply]
Thank you very much.
I don’t want to run mail on VPS entirely - idea is mail is stored privately on my trusted homeserver, VPS is just a forwarder and forgets everything immediately. Also, don’t want to expose LDAP to the VPS.
I thought about using relay service but I think I prefer to self-host it

thanks for the links, they are very helpful. I’ll try to get this working today.

And yes, I do have DNS domain setup. And my VPS provider is unblocked port 25 and the IP is out of spam lists


But I’ve also read that mail servers will retry sending mail, so I could as well have it delivered directly to my homeserver and only use VPS for outbound. I think that might be what I’ll do for less complexity. I can’t send from my homeserver because port 25 is blocked outbound.
